src/OpenRoad.cc

```diff
@@ -437,7 +453,7 @@ void OpenRoad::writeCdl(const char* outFilename,
   }
 }
 
-void OpenRoad::readDb(const char* filename)
+void OpenRoad::readDb(const char* filename, bool hierarchy)
```

Comment:
Can we determine hierarchy directly from the db rather than from the user?  Perhaps having any dbModNet would suffice.

Reply:
Please let us hold off turning on hierarchy by default until we have a little more functionallity working (eg resizer). There are two ways of getting hierarchical designs into the flow: read_verilog -hier and read_db -hier, both explicitly require the -hier flag. Yes for sure we can identify a hierarchical design by finding hierarchical objects as you identify. But I respectfully suggest not doing that until we are confident that the core tools actually support hierarchy.

Reply:
I'm suggesting that if you chose to turn hierarchy on by using -hier then we  should restore in the same mode rather than forcing the users to tell us again.  I'm not proposing that it be on by default in general.

Reply:
Another way to say it: can we tell by inspection that this db must have been built with -hier originally.

Reply:
Possibly some confusion here. The database will be read in fully no matter what. So right now the full hierarchy is there, always and save/restore always work with each other. 

The -hier is simply setting a flag on the dbNetwork so that things like verilog writing and other parts of the code that might do something different with hierarchy will do it. This is a bit like saying: read_db my_db; set hierarchy_flag = true. The -hier is not changing the behaviour of the restore. It is notifying the system (cts, resizer and other prototype code) that it is ok to use the hierarchy. There is no need to tell anyone about the -hier option, the hierarchy is restored by default. But the -hier simply sets a flag that we can inspect in the code to try out support for hierarchy.

Reply:
> There is no need to tell anyone about the -hier option

Its OSS so anybody can see it (eg Peter already tried -hier despite being undocumented).  I don't know what the use model would be where you link -hier but on restore don't want to use hierarchy.  It seems the db already knows enough to decide this without the user needing to give input.

Reply:
Again, all the -hier does (for both read_verilog and read_db) is set a flag saying "ok to try out the hierarchy stuff" (either verilog source or read_db sourced designs). Perhaps one use model is a designer doing a little timing analysis on a hierarchical design (that stuff seems to work now) and possibly choosing the operator realization based on first cut timing analysis of the operator realization and then writing out the design to db. Then later running the design from the db to use the full backend suite resizing etc (that stuff does not work with hierarchy right now).